db/post_migrate/20211201221028_migrate_email_to_normalized_email.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class MigrateEmailToNormalizedEmail < ActiveRecord::Migration[6.1]
+
+  # minimize locking on user_email table
+  disable_ddl_transaction!
+
+  def up
+
+    min, max, total = DB.query_single "SELECT MIN(id), MAX(id), COUNT(*) FROM user_emails"
+    # scaling is needed to compensate for "holes" where records were deleted
+    # and pathological cases where for some reason id 100_000_000 and 0 exist
+
+    # avoid doing any work on empty dbs
+    return if min.nil?
```

Comment:
We need to handle `min == max` as well (i.e. one record). Maybe just set scaling to `1` in that case?

Reply:
oh ... I see ... yikes ... fixing